src/handlers/notify_zulip.rs

```diff
@@ -52,26 +54,71 @@ pub(super) async fn parse_input(
 fn parse_label_change_input(
     event: &IssuesEvent,
     label: Label,
-    config: &NotifyZulipLabelConfig,
+    config: &NotifyZulipNameConfig,
 ) -> Option<NotifyZulipInput> {
-    if !has_all_required_labels(&event.issue, config) {
-        // Issue misses a required label, ignore this event
+    let mut is_default_valid = false;
+    let mut names: Vec<String> = vec![];
+
+    match &config.default {
+        Some(label_config) => {
+            if has_all_required_labels(&event.issue, &label_config) {
+                match event.action {
+                    IssuesAction::Labeled { .. } if !label_config.messages_on_add.is_empty() => {
+                        is_default_valid = true;
+                    }
+                    IssuesAction::Unlabeled { .. }
+                        if !label_config.messages_on_remove.is_empty() =>
+                    {
+                        is_default_valid = true;
+                    }
+                    _ => (),
+                }
+            }
```

Comment:
Can you de-duplicate this code with the one below?

Reply:
I thought it before but there are too many differences in these codes so I think de-duplication doesn't worth it.

Reply:
Taking a step back, most of the duplication is due to "default" and named case being two separate cases the deserialization, maybe we should re-conciliate them at parsing time, as to avoid complicated logic down the line.

What I'm thinking is to pretend that when the user wrote `[notify-zulip."beta-nominated"]` they actually wrote `[notify-zulip."beta-nominated".""]` (notice the empty string).

I experimented with the deserializing of that with serde and a custom Deserializer at https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=aed3e7f8424ca4ea56da6216d06d3e8a.

I'm curious what you think of it and if it helps with the de-duplication in the code.

Reply:
Done.